### PR TITLE
1.x: Do not hide original exception with RxJavaHooks.enableAssemblyTracking()

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
Part of #4212.

### CODE REVIEW

File: src/main/java/rx/internal/operators/OnSubscribeOnAssembly.java [hunk 1]
```diff
@@ -40,15 +41,14 @@
...
-    static String createStacktrace() {
-        StackTraceElement[] stes = Thread.currentThread().getStackTrace();
+    static StackTraceElement[] assemblyStacktrace() {
+        final StackTraceElement[] in = new Exception().getStackTrace();
+        final List<StackTraceElement> out = new ArrayList<StackTraceElement>(in.length);
 
-        StringBuilder sb = new StringBuilder("Assembly trace:");
-        
-        for (StackTraceElement e : stes) {
+        for (StackTraceElement e : in) {
```
Comment (artem-zinnatullin) [inline src/main/java/rx/internal/operators/OnSubscribeOnAssembly.java:51]: I'll open separate PR after this one where I'll move filtering stacktrace to `onError`. This is a hot path while `onError` isn't, so we can save some CPU time on filtering for errorless chains.

File: src/main/java/rx/internal/operators/OnSubscribeOnAssembly.java [hunk 2]
```diff
@@ -30,7 +31,7 @@
 
     final OnSubscribe<T> source;
     
-    final String stacktrace;
```
Comment (akarnokd) [inline src/main/java/rx/internal/operators/OnSubscribeOnAssembly.java:33]: This field had a purpose to show the stacktrace in debug mode by viewing the field as string!

Comment (artem-zinnatullin) [inline src/main/java/rx/internal/operators/OnSubscribeOnAssembly.java:33]: Idk, it consumes memory and I've never used it during debugging. You really need it? I think you should be able to get ~normal output from `Arrays.toString(assemblyStacktrace)` in the debugger

**Comment by akarnokd** [comment]
See #4215 for a less invasive change.

**Comment by artem-zinnatullin** [comment]
Closing in favor of #4215.

**Pull request closed without merging**